### ISSUE TITLE
lost 20% of my hashrate after Monero fork

### ISSUE THREAD

**Issue by tinkyisretarded** [fireice-uk/xmr-stak#1964]
Pre fork 6xVega56 = 11300H/s
Post fork 6xVega56 = 9100H/s

**Comment by keefstone** [comment]
Bumping the worksize from 8 to 16 [seems to be working](https://github.com/fireice-uk/xmr-stak/issues/1962#issuecomment-431016582).

**Comment by Spudz76** [comment]
Everyone is losing 5-12% on the fork, thus nobody loses really.  You will still have similar share of the total network hashrate, because the whole network hashrate is going to drop by at least 15% in my prediction.

It does shake up what the best XMR mining hardware is.  What was best for v7 may suck ass at v8, oh well.  A few things do v8 faster than v7 somehow, those are the new best mining devices.  Sorry for the shakeup, but it sure beats ASIC nicely.

**Comment by blah415** [comment]
thanks, that worksize helped. my rx580s lose absolute zero hashrate, so those mining Rxs have lost nothing. only vega rigs.

**Comment by gitmanfa** [comment]
why also increase in power consumption  by 10% ? thanks for wonderful software

**Comment by Spudz76** [comment]
Harder work in new algo, so it is either slower or will use more watts.  Integer square root and long integer division in one step - which ASIC can never ever handle except at 4x-12x speed penalty, which makes them as 'slow' as a GPU at more power usage (aka unprofitable, so game over).

To do the new math in one step simultaneously you have to have predictive side execution like a true full processor, and ASIC can't do that.  Old CPUs and some GPUs tend to really suck at the new math too due to missing multipath execution or crap integer performance, sorry, oh well.  Worth it to keep out ASIC forever.

I also have disabled all Spectre/Meltdown and friends mitigations wherever possible to ensure nothing is blocking the parallelism and maximum (even if insecure) memory sharing opportunities.  Not sure if it actually helps but having less garbage in the way can't be worse.

**Comment by leve1ord** [comment]
Yep, this fork gives less hash and more power consuption and heat. Nice fork (no).
R5 1600 then 600h/ now 550h
RX 480 - 980h/920h
RX 580 - 975h/890h
RX 560 - 480h/460

**Comment by blah415** [comment]
not true for rx580 msi armor, i haven't lost hash rate, im getting about 994 hash

**Comment by gotminer** [comment]
@TheGoddessInari Yeah I'm running out of idea's ... I've switched to the two thread per gpu code in the doc's section.  Toggled work size back and forth, adjusted intensity to 1792.

**Comment by gotminer** [comment]
Should mem_chunk be 2 for Vega 56's?

**Comment by blah415** [comment]
yes mem_chunk 2 for vegas.. only thing changed was worksize to 16 and im getting about 1800-1850 per vega instead of 1900-1920(v7)

**Comment by gotminer** [comment]
@blah415 Damn ... I'm only getting around 1000 per Vega since the fork this morning on 2.5.0 and 2.5.1.

**Comment by psychocrypt** [comment]
in windows change the driver to 18.5.1 or older. The newer driver are
broken. I tried also the today released driver but this is also not working.

**Comment by blah415** [comment]
@gotminer try different drivers

**Comment by gotminer** [comment]
@psychocrypt Specifically does the blockchain driver from Aug 2017 not work?  That's the one I am using.  Seems that 18.5.1 came out in April 2018.

**Comment by psychocrypt** [comment]
other users reported that the blockchaindriver also work. But I have no
vega with windows.

**Comment by gotminer** [comment]
Might be onto something here.  I just used ddu to remove the blockchain drivers, installed Adrenalin Edition 18.5.1 drivers.  Setup xmr-stak from scratch and I have half of my hash rate without adding the second thread onto each gpu.  Going to edit the amd file and see what I come up with.

**Comment by TheGoddessInari** [comment]
I've tried on Windows 18.5.1, 18.9.2, and on Linux 18.10 and ROCm 1.9.1, all results are the same on the same machine. No invalid hashes, just half the usual hashrate of v7 and no settings changes will bring it back to normal.

There's http://www.amdsurveys.com/se/5A1E27D23A3DE966 for Windows and https://github.com/RadeonOpenCompute/ROCm/issues for Linux as far as OpenCL issue reporting.

**Comment by xavierx66** [comment]
The blockchain driver does not work for me anymore with xmr-stak 2.5.0

Using driver 18.5.1 now works with v8, but the hashrate is 50% on RX cards

Vega's I have working fine with 18.5.1, but RX cards run terrible

**Comment by gotminer** [comment]
@xavierx66 Will you post the contents of your amd file for the Vega's?

**Comment by TheGoddessInari** [comment]
Also appears to only affect xmr-stak-based miners, and only V8. Both the "good" opencl compiler and the "bad" opencl compiler are producing the same awful results.

**Comment by xavierx66** [comment]
"gpu_threads_conf" : [
  // gpu: gfx900 memory:3920
  // compute units: 56
  { "index" : 0,
    "intensity" : 1792, "worksize" : 16,
    "affine_to_cpu" : false, "strided_index" : 1, "mem_chunk" : 2,
    "unroll" : 8, "comp_mode" : true
  },
  { "index" : 0,
    "intensity" : 1792, "worksize" : 16,
    "affine_to_cpu" : false, "strided_index" : 1, "mem_chunk" : 2,
    "unroll" : 8, "comp_mode" : true
  },

**Comment by gotminer** [comment]
It's getting better ... Still have two gpu's with half the hash rate that they should have though.

**Comment by poteus** [comment]
Vega 56: 1600H/s with drivers 18.5.1... (lost 250H/s but I have not tried to tweak it yet).

**Comment by poteus** [comment]
HASHRATE REPORT - AMD
| ID |    10s |    60s |    15m | ID |    10s |    60s |    15m |
|  0 |  833.1 |   (na) |   (na) |  1 |  830.4 |   (na) |   (na) |
|  2 |  834.0 |   (na) |   (na) |  3 |  828.1 |   (na) |   (na) |
Totals (AMD):  3325.6    0.0    0.0 H/s

**Comment by gotminer** [comment]
@TheGoddessInari Are you losing hash rate over all gpu's or just certain ones?

**Comment by TheGoddessInari** [comment]
@gotminer Over both GPUs, all threads evenly. It's ~1050 per thread on v7, ~500 per thread on v8, regardless of settings.

**Comment by poteus** [comment]
|  0 |  855.4 |   (na) |   (na) |  1 |  851.4 |   (na) |   (na) |
|  2 |  852.0 |   (na) |   (na) |  3 |  859.5 |   (na) |   (na) |

That's my best with Vega 56.  Just changed "strided_index" : 2...

What's this tho?
[2018-10-18 19:15:10] : AMD Invalid Result GPU ID 1

**Comment by gotminer** [comment]
I just tried cast-xmr and I'm getting a decent hash rate.  7% drop from what I was getting with xmr-stak on monero7.  I'm going to say that's reasonable with stock settings.  I just don't understand why I can't seem to get it out of xmr-stak.
